executor/tests/test_executor.py

```diff
@@ -630,12 +613,7 @@ def assertions(driver, task_id, sandbox_directory):
             assert_message(self, expected_message_1, actual_encoded_message_1)
 
         stop_signal = Event()
-
-        def sleep_and_set_stop_signal():
-            time.sleep(2 * cook.RUNNING_POLL_INTERVAL_SECS)
-            stop_signal.set()
-        thread = Thread(target=sleep_and_set_stop_signal, args=())
-        thread.start()
+        Thread(target=sleep_and_set_stop_signal, args=(stop_signal, 2 * cook.RUNNING_POLL_INTERVAL_SECS)).start()
```

Comment:
Timer?

You could actually move the creation of the Timer into a helper method so that the implementation doesn't leak through at all of these call-sites. (You'd want to use `async` in the name to make it clear that it doesn't block the current thread.)

Reply:
Created `sleep_and_set_stop_signal_task`. Please see above.